### PR TITLE
Add entities answers popover in card header (#7230)

### LINKED ISSUE #7230: Number of entities in the header of a "smart" notification
Hi OpFab colleagues,

We noticed that currently in the header of the smart notifications (notifications in which you can provide an answer), the number of displayed entity is limited to 3 and the others are in "..." via a clickable tool-tip:

image

Would it be possible to extend this number of displayed entity ? 6 would be sufficient for our use-case.

Thanks in advance.

Kind regards,

### PR DESCRIPTION
**Pull request by quinarygio**
Fix #7230 

- In release notes :
  -  In chapter : Features 
  -  Text : #7230 : Add entities answers popover in card header 

## Summary by CodeRabbit

- **New Features**
	- Enhanced display of entities in the card header with a new table format for answered and not answered lists.
	- Improved interaction model for opening the entity dropdown in the card header.

- **Bug Fixes**
	- Streamlined visibility assertions for entities, ensuring accurate representation of their status.

- **Style**
	- Updated styling for the dropdown to manage text overflow and enhance visual presentation.
	- Introduced a new CSS class for adjusting popover arrow positioning.

- **Refactor**
	- Simplified logic for handling entity lists, consolidating properties and methods for better clarity and efficiency.

### CODE REVIEW

File: ui/main/src/app/modules/card/components/card-header/card-header.component.scss [hunk 1]
```diff
@@ -10,19 +10,10 @@
 .opfab-entities-dropdown {
     text-align: center;
     padding-left: 5px;
+    white-space: nowrap;
+    text-overflow: ellipsis; 
+    overflow: hidden;
```
[the remark this answers is not in the document]
Comment (quinarygio) [reply ui/main/src/app/modules/card/components/card-header/card-header.component.scss:15]: Not needed as there is already a popover wich shows the full list of entities

File: ui/main/src/app/modules/card/components/card-header/card-header.component.ts [hunk 2]
```diff
@@ -85,9 +83,9 @@ export class CardHeaderComponent implements OnChanges {
     }
 
     private setEntitiesForCardHeader(entities) {
-        const entitiesForCardHeader = this.getEntitiesForCardHeaderFromEntityIds(entities);
-        entitiesForCardHeader.sort((a, b) => a.name?.localeCompare(b.name));
-        this.separateEntitiesBetweenVisibleAndDropdown(entitiesForCardHeader);
+        this.entitiesForCardHeader = this.getEntitiesForCardHeaderFromEntityIds(entities);
+        this.entitiesForCardHeader.sort((a, b) => a.name?.localeCompare(b.name));
+        this.computeAnswersLists();
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/modules/card/components/card-header/card-header.component.ts:91]: it is a detail and it is subject to discussion , do not need to take into account

File: ui/main/src/app/modules/card/components/card-header/card-header.component.html [hunk 3]
```diff
@@ -46,9 +41,23 @@
...
\ No newline at end of file
+      <table>
+          <tr>
+              <th style="padding:10px"><span translate>response.answered</span></th>
+              <th style="padding:10px"><span translate>response.unanswered</span></th>
+          </tr>
+          <tr>
+              <td id="opfab-answered-list" style="vertical-align:top; padding:10px">
+                  <div *ngFor="let entity of answeredList">
+                      <span style="white-space: nowrap;" [ngStyle]="{'color': entity.color}" [id]="'opfab-list-aanswered-entity-'+entity.id">&nbsp; {{entity.name}} &nbsp;</span>
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/card/components/card-header/card-header.component.html]: aanswered --> answered

File: ui/main/src/app/modules/card/components/card-header/card-header.component.html [hunk 4]
```diff
@@ -46,9 +41,23 @@
...
+                  </div> 
+              </td>
+              <td id="opfab-not-answered-list" style="vertical-align:top; padding:10px">
+                  <div *ngFor="let entity of notAnsweredList">
+                      <span style="white-space: nowrap;" [ngStyle]="{'color': entity.color}" [id]="'opfab-list-not-aanswered-entity-'+entity.id">&nbsp; {{entity.name}} &nbsp;</span>
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/card/components/card-header/card-header.component.html]: aanswered --> answered

**Comment by freddidierRTE** [comment]
Center the popup on the text (the list of entities) if possible 

image

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

**Review by freddidierRTE — APPROVED** [review]

**Pull request merged**